src/server/snap.rs

```diff
@@ -394,6 +395,9 @@ fn recv_snap<R: RaftExtension + 'static>(
                 return Err(e);
             }
         }
+        // Notify the snapshot manager that a snapshot has been received,
+        // freeing up the associated resource in the concurrency limiter.
+        snap_mgr.recv_snap_complete(context.raft_msg.region_id);
```

Comment:
Maybe let it complete after the snapshot is applied

Reply:
Updated it as suggested